extensions/amp-story/1.0/amp-story.js

```diff
@@ -1016,6 +1016,7 @@ export class AmpStory extends AMP.BaseElement {
       .then(() => {
         this.markStoryAsLoaded_();
         this.initializeLiveStory_();
+        this.initializeStoryPlayer_();
```

Comment:
What's the timeout before the `amp-story-player` and its messaging API throws and error because the AMP document didn't respond to the handshake?

What happens on the `amp-story-player` side if the `amp-story.js` script takes a few seconds to load, and then the `this.whenPagesLoaded_(PAGE_LOAD_TIMEOUT_MS))` times out?

Reply:
> What's the timeout before the amp-story-player and its messaging API throws and error because the AMP document didn't respond to the handshake?

[20000 ms](https://github.com/ampproject/amphtml/blob/e3be1b2426e07fa27681d3139e2fc96e56974e44/src/service/viewer-impl.js#L306)

> What happens on the amp-story-player side if the amp-story.js script takes a few seconds to load, and then the this.whenPagesLoaded_(PAGE_LOAD_TIMEOUT_MS)) times out?

While the script is loading we would show the loader/poster image of the first story, and the story will be displayed when the script is loaded. If `this.whenPagesLoaded_(PAGE_LOAD_TIMEOUT_MS))` it would display an error in the console (which we haven't set yet), but it doesn't block laying out the story.

Reply:
Sorry, my question was about the `amp-story-player`. It was triggering some errors if the `amp-viewer-integration` script was not loaded, wouldn't it trigger the same errors if the `amp-viewer-integration` is loaded too late?
What's the timeout on the player side of the viewer, before it throws one of these errors?

Reply:
Oh I see what you mean... Yeah it will throw if no messaging channel has been initialized in 20s...

Were you thinking of adding additional an additional timeout at the player level?

Reply:
If the story doesn't load in 20s we have a problem, that sgtm :)) Thanks for investigating